netplan/cli/commands/status.py

```diff
@@ -0,0 +1,640 @@
+                    pprint(('{title:>'+pad+'} {value}').format(
+                        title='DNS Addresses:' if i == 0 else '',
+                        value=val,
+                        ))
+
+            lst = data.get('dns_search', [])
+            if len(lst) > 0:
+                for i, val in enumerate(lst):
+                    pprint(('{title:>'+pad+'} {value}').format(
+                        title='DNS Search:' if i == 0 else '',
+                        value=val,
+                        ))
+
+            lst = data.get('routes', [])
+            if len(lst) > 0:
+                for i, obj in enumerate(lst):
+                    default_start = ''
+                    default_end = ''
+                    if obj['to'] == 'default':
+                        default_start = '[highlight]'
+                        default_end = '[/highlight]'
+                    via = ''
+                    if 'via' in obj:
+                        via = ' via ' + obj['via']
+                    src = ''
+                    if 'from' in obj:
+                        src = ' from ' + obj['from']
+                    metric = ''
+                    if 'metric' in obj:
+                        metric = ' metric ' + str(obj['metric'])
+
+                    extra = []
+                    if 'protocol' in obj and obj['protocol'] != 'kernel':
+                        proto = obj['protocol']
+                        extra.append(proto)
+                    if 'scope' in obj and obj['scope'] != 'global':
+                        scope = obj['scope']
+                        extra.append(scope)
+                    if 'type' in obj and obj['type'] != 'unicast':
+                        type = obj['type']
+                        extra.append(type)
+
+                    pprint(('{title:>'+pad+'} {start}{to}{via}{src}{metric}{end}[muted]{extra}[/muted]').format(
+                        title='Routes:' if i == 0 else '',
+                        to=obj['to'],
+                        via=via,
+                        src=src,
+                        metric=metric,
+                        extra=' ('+', '.join(extra)+')' if len(extra) > 0 else '',
+                        start=default_start,
+                        end=default_end))
+
+            val = data.get('activation_mode')
+            if val:
+                pprint(('{title:>'+pad+'} {value}').format(
+                    title='Activation Mode:',
+                    value=val,
+                    ))
+            pprint()
+
+        hidden = total - len(interfaces)
+        if (hidden > 0):
+            pprint('{} inactive interfaces hidden. Use "--all" to show all.'.format(hidden))
+
+    def command(self):
+        # required data: iproute2 and sd-networkd can be expected to exist,
+        # due to hard package dependencies
+        iproute2 = self.query_iproute2()
+        networkd = self.query_networkd()
+        if not iproute2 or not networkd:
```

Comment:
systemd[-networkd] should be installed on any relevant system, as it is a dependency of netplan.io. IIRC I designed this new CLI in a way that it would assume networkd to be available... But maybe I should double-check if networkd is actually up and running on a system which does not necessarily depend on it (like a standard Ubuntu Desktop system).

Reply:
Dang. I can confirm that systemd-networkd is installed but not running on a default Ubuntu Desktop installation. Let's see what I can do about that (maybe starting the service before trying to query it).

Reply:
Checking the activity status of `systemd-networkd.service` and starting it if needed solves this issue. We assume systemd-networkd to be installed by default (it's part of the `systemd` package in Ubuntu).

One drawback: `netplan status` would be able to run without sudo privileges, if not for this reason of starting a system service... But maybe that's something to solve at a later time, as Netplan is a system/admin tool and requires root privileges for most of it's actions anyways.

Reply:
Right. So, assuming sd-networkd is a hard dependency for netplan status, and I don't have a strong opinion on that, we should check first if we can start it. If it's masked for example netplan status will crash.

It would be nice to show a nice message like "netplan status depends on networkd" if we can't start it. A good thing is that once it's running one can call netplan status as a non-privileged user.

Reply:
Or maybe we could just tell the user that networkd is not running and ask them to start it instead of trying to do that inside netplan...

Reply:
I'd like to have it working "out of the box" on a default Ubuntu Desktop installation, that's why I'd like to automatically have it started if needed.

But checking if systemd-networkd.service is masked is a fair point. We'll now be bailing out if networkd is masked with an error message notifying the user to start networkd. If it's just disabled OTOH we'll try to start it automatically.